### DISCUSSION TITLE
Great progress. Help on collecting and writing up the changelog 🚀

### DISCUSSION THREAD

**Discussion by AntonOsika** [AntonOsika/gpt-engineer#125] [Announcements]
We have done quite some improvements since the weekend started 🚀🤖

On our path towards singularity, achieving our milestone of:

_Bootstrapping GPT Engineer to let the right prompt, generate a better version of GPT Engineer itself_

### Why a changelog?
We want to show to ourselves and the outside world the progress we are doing.

Hence: A changelog.

**Ask:** Would anyone be interested in owning the release of our changelog?

### Some examples of things we can already put in it and make a great announcement:

**fast feedback loops** 🏎️
- Automatic execution for fast feedbackloop
- Making the interactive clarification step optional

**Contributor ergonomics** 🍦
- Configurable "steps" in the generation process for fast iteration
- Autoformatting, pre-commit hooks, pyproject configurations

**Improved performance** ⚙️
- Generation of specification before it starts
- Self-reflection
- Test Driven Development workflows to guarantee stability

**Benchmarks** 🤼‍♀️
- 1. GAME IS ON. We now have a benchmark of to evaluate our progress, and be able to prove that GPT Engineer is the best virtual programmer. Any other project are welcome to try to beat us on this.

Send [@AntonOsika](https://twitter.com/antonosika) a message if you are interested in writing the changelog, together with an example link of what format you would like to use.

**Comment by noxiouscardiumdimidium** [comment]
my pascal script might help with that, by generating more pure code out put, with minimal but meaningful comments/explanations unless asked... if your pipeline could push a "continue generating" you could get 15k context, concise, collapsed code off one-shots... the PASCAL persona is so code fixated, it doesnt say hello if you load it without an init prompt footer... it just displays its traits and coding skills in the form of a markdown codeblock... do a test run, PRIOR to the satandard workflow you have now, without adding additional formatting, appended with the philosophy text, and main prompt - then imediatley follow up with the current build process

chatgpt chat link: PASCAL+Philosophy+main prompt one-shot output

https://chat.openai.com/share/17636bf2-f5f3-400e-905a-9e457e3462a9
